common/sagas/swap/orders.js

```diff
@@ -0,0 +1,166 @@
+  try {
+    yield put(stopLoadBityRatesSwap());
+    const order = yield call(
+      postOrder,
+      payload.amount,
+      payload.destAddress,
+      payload.mode,
+      payload.pair
+    );
+    if (order.error) {
+      // TODO - handle better / like existing site?
+      yield put(showNotification('danger', `Bity Error: ${order.msg}`, 10000));
+      yield put({ type: 'SWAP_ORDER_CREATE_FAILED' });
+    } else {
+      yield put(orderCreateSucceededSwap(order.data));
+      yield put(changeStepSwap(3));
+      // start countdown
+      yield put(startOrderTimerSwap());
+      // start bity order status polling
+      yield put(startPollBityOrderStatus());
+    }
+  } catch (e) {
+    const message =
+      'Connection Error. Please check the developer console for more details and/or contact support';
+    yield put(showNotification('danger', message, TEN_SECONDS));
+    yield put({ type: 'SWAP_ORDER_CREATE_FAILED' });
+  }
+}
+
+export function* postBityOrderSaga(): Generator<Effect, void, any> {
+  yield takeEvery('SWAP_ORDER_CREATE_REQUESTED', postOrderCreate);
+}
+
+export function* bityTimeRemaining() {
+  while (yield take('SWAP_ORDER_START_TIMER')) {
+    let hasShownNotification = false;
+    while (true) {
+      yield call(delay, ONE_SECOND);
+      const swap = yield select(getSwap);
+      // if (swap.bityOrder.status === 'OPEN') {
+      const createdTimeStampMoment = moment(
+        swap.orderTimestampCreatedISOString
+      );
+      let validUntil = moment(createdTimeStampMoment).add(swap.validFor, 's');
+      let now = moment();
+      if (validUntil.isAfter(now)) {
+        let duration = moment.duration(validUntil.diff(now));
+        let seconds = duration.asSeconds();
+        yield put(orderTimeSwap(parseInt(seconds)));
+        // TODO (!Important) - check orderStatus here and stop polling / show notifications based on status
+      } else {
+        switch (swap.orderStatus) {
+          case 'OPEN':
+            yield put(orderTimeSwap(0));
+            yield put(stopPollBityOrderStatus());
+            yield put({ type: 'SWAP_STOP_LOAD_BITY_RATES' });
+
+            if (!hasShownNotification) {
+              hasShownNotification = true;
+              yield put(showNotification('danger', BITY_TIMEOUT_MESSAGE, 0)); // 0 is forever
+            }
+            break;
+          case 'CANC':
+            yield put(orderTimeSwap(0));
+            yield put(stopPollBityOrderStatus());
+            yield put({ type: 'SWAP_STOP_LOAD_BITY_RATES' });
+            if (!hasShownNotification) {
+              hasShownNotification = true;
+              yield put(showNotification('danger', BITY_TIMEOUT_MESSAGE, 0)); // 0 is forever
+            }
+            break;
+          case 'RCVE':
+            yield put(orderTimeSwap(0));
+            if (!hasShownNotification) {
+              hasShownNotification = true;
+              yield put(showNotification('warning', BITY_TIMEOUT_MESSAGE, 0)); // 0 is forever
```

Comment:
Given the explanation added to all 3, maybe we should change the showNotification api to take in `infinity` to be a bit more "readable"